### PR TITLE
Test Longer Tests in Travis

### PR DESCRIPTION
**Pull request by Gouthamve**
Don't know why this was not being done before.

cc @juliusv @grobie

### CODE REVIEW

**Comment by brancz** [comment]
lgtm

**Comment by juliusv** [comment]
I believe we run all tests already on CircleCI (takes longer than TravisCI, but is a more complete feedback): https://github.com/prometheus/prometheus/blob/master/circle.yml#L34-L36

See also https://quay.io/repository/prometheus/golang-builder?tag=1.8-base

```
Usage: builder.sh [args]
  -i,--import-path arg  : Go import path of the project
  -p,--platforms arg    : List of platforms (GOOS/GOARCH) to build separated by a space
  -T,--tests            : Go run tests then exit
```

**Comment by Gouthamve** [comment]
Oh wow, we don't run all the tests. See: https://github.com/prometheus/golang-builder/blob/master/1.8/base/rootfs/common.sh#L74

Even Circle runs only short tests. So if I have to fix this in circle also, `test` target should be running longer tests too. Should I make that change?

**Comment by beorn7** [comment]
I think the history here is that the long tests could run so long at times that they would time out in the CI pipeline.
Thanks to performance improvements over time, the long tests don't run for that long anymore, so we could run all tests for now.

**Comment by Gouthamve** [comment]
Okay, just to be clear, I will remove the `testall` and make the `test` target itself run all the tests.

I did run all tests on my system and the results are not as flaky.

**Comment by juliusv** [comment]
I'd be for having both targets and only running the short tests locally, but the long ones in CI.

The short ones take 32s for me and the long ones 52s. Keep the local testing short, but have a longer safety net before merging :)

**Comment by Gouthamve** [comment]
@juliusv Added new target for shorter tests as Circle uses a base image that I think is shared across all projects in the Prometheus org. In that base image `make test` is run.

File: Makefile [hunk 1]
```diff
@@ -36,10 +36,15 @@ check_license:
...
-test:
+test-short:
 	@echo ">> running short tests"
 	@$(GO) test -short $(pkgs)
 
+test:
+	@echo ">> running all tests"
+	@$(GO) test $(pkgs)
+
+
```
Comment (juliusv) [inline Makefile]: Remove extra empty line.

**Comment by juliusv** [comment]
@Gouthamve Huh, you're right. See, that's how little I know about our testing/CI setup :)

Just one nit comment, otherwise :+1:

**Comment by Gouthamve** [comment]
Done. Sorry, did not notice it before.

**Pull request merged**